internal/controller/repository/repository_collection.go

```diff
@@ -0,0 +1,141 @@
+package repository
+
+import (
+	"fmt"
+
+	"github.com/kyma-project/helm-broker/internal"
+	addonsv1alpha1 "github.com/kyma-project/helm-broker/pkg/apis/addons/v1alpha1"
+)
+
+// Collection keeps and process collection of Repository
+type Collection struct {
+	Repositories []*Repository
+}
+
+// NewRepositoryCollection returns pointer to RepositoryCollection
+func NewRepositoryCollection() *Collection {
+	return &Collection{
+		Repositories: []*Repository{},
+	}
+}
+
+// AddRepository adds new Repository to RepositoryCollection
+func (rc *Collection) AddRepository(repo *Repository) {
+	rc.Repositories = append(rc.Repositories, repo)
+}
+
+func (rc *Collection) addons() []*Entry {
+	var addons []*Entry
+
+	for _, repo := range rc.Repositories {
+		for _, addon := range repo.Addons {
+			addons = append(addons, addon)
+		}
+	}
+
+	return addons
+}
+
+func (rc *Collection) completeAddons() []*Entry {
+	var addons []*Entry
+
+	for _, addon := range rc.addons() {
+		if !addon.IsComplete() {
+			continue
+		}
+		addons = append(addons, addon)
+	}
+
+	return addons
+}
+
+// ReadyAddons returns all addons from all repositories which ready status
+func (rc *Collection) ReadyAddons() []*Entry {
+	var addons []*Entry
+
+	for _, addon := range rc.addons() {
+		if !addon.IsReady() {
+			continue
+		}
+		addons = append(addons, addon)
+	}
+
+	return addons
+}
+
+// IsRepositoriesFailed informs if any of repositories in collection is in failed status
+func (rc *Collection) IsRepositoriesFailed() bool {
+	for _, repository := range rc.Repositories {
+		if repository.IsFailed() {
+			return true
+		}
+		if repository.HasFailedAddons() {
+			repository.Failed()
```

Comment:
If the first repository will be failed then method returns without changing other repositories status even if the next ones are failed

Reply:
fixed